update.sh uses BASH_SOURCE to determine directory

### Issue details

In [update.sh in line 10 ](https://github.com/wallabag/wallabag/blob/master/scripts/update.sh#L10) we use BASH_SOURCE to try to find the directory of update.sh script:

```bash
DIR="${BASH_SOURCE}"
```

But according to documentation, this variable is an array.

> BASH_SOURCE
>
>    An array variable whose members are the source filenames where the corresponding shell function names in the FUNCNAME array variable are defined. The shell function ${FUNCNAME[$i]} is defined in the file ${BASH_SOURCE[$i]} and called from ${BASH_SOURCE[$i+1]}

https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#index-BASH_005fSOURCE


If we want to find the directory of the update.sh script, we should follow a different approach, e.g.:

```bash
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

https://stackoverflow.com/a/246128/709697


### Environment

* wallabag version (or git revision) that exhibits the issue: 2.3.2
* How did you install wallabag? Via `git clone` or by downloading the package?git
* Last wallabag version that did not exhibit the issue (if applicable):2.3.1
* php version:
* OS:debian
* type of hosting (shared or dedicated):dedicated
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL):postgresql

I suppose I already saw an issue resulting from this. While trying to upgrade the setup found a missing PHP extension: 

> roots@roots-server /var/www/x22889/wallabag $ sudo make update
> [sudo] password for roots:
> composer.phar not found, we'll see if composer is installed globally.
> remote: Counting objects: 777, done.
> remote: Compressing objects: 100% (69/69), done.
> remote: Total 777 (delta 456), reused 499 (delta 444), pack-reused 255
> Receiving objects: 100% (777/777), 154.48 KiB | 0 bytes/s, done.
> Resolving deltas: 100% (510/510), completed with 151 local objects.
> From https://github.com/wallabag/wallabag
>  * [new branch]      2.4        -> origin/2.4
>  * [new branch]      add-random-article -> origin/add-random-article
>  + e4dcc1b...9d5f120 change-locale-register -> origin/change-locale-register  (forced update)
>  * [new branch]      fix-migrations-dash -> origin/fix-migrations-dash
>    cabadf1..f24e9bf  master     -> origin/master
>  * [new branch]      php-7.2.1  -> origin/php-7.2.1
>  + c79b60c...e2a48a6 store-resolved-url -> origin/store-resolved-url  (forced update)
>  * [new branch]      url-3529   -> origin/url-3529
> remote: Counting objects: 6, done.
> remote: Compressing objects: 100% (2/2), done.
> remote: Total 6 (delta 3), reused 5 (delta 3), pack-reused 1
> Unpacking objects: 100% (6/6), done.
> From https://github.com/wallabag/wallabag
>  * [new tag]         2.3.1      -> 2.3.1
>  * [new tag]         2.3.2      -> 2.3.2
> Previous HEAD position was 79ad373... Release wallabag 2.3.0
> HEAD is now at b1b4ae6... Release wallabag 2.3.2
> Running composer as root/super user is highly discouraged as packages, plugins and scripts cannot always be trusted
> Loading composer repositories with package information
> Installing dependencies from lock file
> Your requirements could not be resolved to an installable set of packages.
> 
>   Problem 1
>     - php-amqplib/php-amqplib v2.7.0 requires ext-bcmath * -> the requested PHP extension bcmath is missing from your system.
>     - php-amqplib/php-amqplib v2.7.0 requires ext-bcmath * -> the requested PHP extension bcmath is missing from your system.
>     - Installation request for php-amqplib/php-amqplib v2.7.0 -> satisfiable by php-amqplib/php-amqplib[v2.7.0].
> 
>   To enable extensions, verify that they are enabled in those .ini files:
>     - /etc/php/7.1/cli/php.ini
>     - /etc/php/7.1/cli/conf.d/10-mysqlnd.ini
>     - /etc/php/7.1/cli/conf.d/10-opcache.ini
>     - /etc/php/7.1/cli/conf.d/10-pdo.ini
>     - /etc/php/7.1/cli/conf.d/15-xml.ini
>     - /etc/php/7.1/cli/conf.d/20-apcu.ini
>     - /etc/php/7.1/cli/conf.d/20-apcu_bc.ini
>     - /etc/php/7.1/cli/conf.d/20-calendar.ini
>     - /etc/php/7.1/cli/conf.d/20-ctype.ini
>     - /etc/php/7.1/cli/conf.d/20-curl.ini
>     - /etc/php/7.1/cli/conf.d/20-dom.ini
>     - /etc/php/7.1/cli/conf.d/20-exif.ini
>     - /etc/php/7.1/cli/conf.d/20-fileinfo.ini
>     - /etc/php/7.1/cli/conf.d/20-ftp.ini
>     - /etc/php/7.1/cli/conf.d/20-gd.ini
>     - /etc/php/7.1/cli/conf.d/20-gettext.ini
>     - /etc/php/7.1/cli/conf.d/20-iconv.ini
>     - /etc/php/7.1/cli/conf.d/20-igbinary.ini
>     - /etc/php/7.1/cli/conf.d/20-json.ini
>     - /etc/php/7.1/cli/conf.d/20-mbstring.ini
>     - /etc/php/7.1/cli/conf.d/20-mcrypt.ini
>     - /etc/php/7.1/cli/conf.d/20-mysqli.ini
>     - /etc/php/7.1/cli/conf.d/20-pdo_mysql.ini
>     - /etc/php/7.1/cli/conf.d/20-pdo_sqlite.ini
>     - /etc/php/7.1/cli/conf.d/20-phar.ini
>     - /etc/php/7.1/cli/conf.d/20-posix.ini
>     - /etc/php/7.1/cli/conf.d/20-readline.ini
>     - /etc/php/7.1/cli/conf.d/20-redis.ini
>     - /etc/php/7.1/cli/conf.d/20-shmop.ini
>     - /etc/php/7.1/cli/conf.d/20-simplexml.ini
>     - /etc/php/7.1/cli/conf.d/20-soap.ini
>     - /etc/php/7.1/cli/conf.d/20-sockets.ini
>     - /etc/php/7.1/cli/conf.d/20-sqlite3.ini
>     - /etc/php/7.1/cli/conf.d/20-sysvmsg.ini
>     - /etc/php/7.1/cli/conf.d/20-sysvsem.ini
>     - /etc/php/7.1/cli/conf.d/20-sysvshm.ini
>     - /etc/php/7.1/cli/conf.d/20-tokenizer.ini
>     - /etc/php/7.1/cli/conf.d/20-wddx.ini
>     - /etc/php/7.1/cli/conf.d/20-xmlreader.ini
>     - /etc/php/7.1/cli/conf.d/20-xmlwriter.ini
>     - /etc/php/7.1/cli/conf.d/20-xsl.ini
>     - /etc/php/7.1/cli/conf.d/20-zip.ini
>   You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
> 
> 
>   [Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException]
>   You have requested a non-existent parameter "database_driver_class". Did you mean this: "database_driver"?
> 
> 
> 
> 
>   [Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException]
>   You have requested a non-existent parameter "database_driver_class". Did you mean this: "database_driver"?
> 
> 
> Makefile:16: recipe for target 'update' failed
> make: *** [update] Error 1

After installing it the setup only returns an error:

> roots@roots-server /var/www/x22889/wallabag $ sudo make update
> scripts/update.sh: 10: scripts/update.sh: BASH_SOURCE: parameter not set
> Makefile:16: recipe for target 'update' failed
> make: *** [update] Error 2

@Strubbl thank you for ```sudo -u www-data bash scripts/update.sh prod```. I was struggling with setting BASH_SOURCE variable with no luck. Used diffrent combinations od sudo -u www-data sh/zsh/bash set/export commands, nothing worked.

```shell
scripts/update.sh: 10: scripts/update.sh: BASH_SOURCE: parameter not set
Makefile:16: recipe for target 'update' failed
make: *** [update] Error 2
```

--- a/scripts/dev.sh
+++ b/scripts/dev.sh
@@ -3,10 +3,14 @@
 # eg: `sh dev.sh`
 
 COMPOSER_COMMAND='composer'
+REQUIRE_FILE='scripts/require.sh'
 
-DIR="${BASH_SOURCE}"
-if [ ! -d "$DIR" ]; then DIR="$PWD/scripts"; fi
-. "$DIR/require.sh"
+if [ ! -f "$REQUIRE_FILE" ]; then
+  echo "Cannot find $REQUIRE_FILE"
+  exit 1
+fi
+
+. "$REQUIRE_FILE"
 
 $COMPOSER_COMMAND install
 php bin/console wallabag:install

--- a/scripts/install.sh
+++ b/scripts/install.sh
@@ -25,10 +25,14 @@ if [ "$IGNORE_ROOT" -eq 0 ] && [ "$EUID" == "0" ]; then
 fi
 
 COMPOSER_COMMAND='composer'
+REQUIRE_FILE='scripts/require.sh'
 
-DIR="${BASH_SOURCE}"
-if [ ! -d "$DIR" ]; then DIR="$PWD/scripts"; fi
-. "$DIR/require.sh"
+if [ ! -f "$REQUIRE_FILE" ]; then
+  echo "Cannot find $REQUIRE_FILE"
+  exit 1
+fi
+
+. "$REQUIRE_FILE"
 
 TAG=$(git describe --tags $(git rev-list --tags --max-count=1))
 

--- a/scripts/update.sh
+++ b/scripts/update.sh
@@ -28,10 +28,14 @@ set -e
 set -u
 
 COMPOSER_COMMAND='composer'
+REQUIRE_FILE='scripts/require.sh'
 
-DIR="${BASH_SOURCE}"
-if [ ! -d "$DIR" ]; then DIR="$PWD/scripts"; fi
-. "$DIR/require.sh"
+if [ ! -f "$REQUIRE_FILE" ]; then
+  echo "Cannot find $REQUIRE_FILE"
+  exit 1
+fi
+
+. "$REQUIRE_FILE"
 
 rm -rf var/cache/*
 git fetch origin

